content/extending-chat-widget/custom-identity-provider/index.mdx

```diff
@@ -0,0 +1,180 @@
+---
+weight: 60
+category: "extending-chat-widget"
+subcategory: "custom-identity-provider"
+title: "Custom Identity Provider"
+tagline: "Pair your own customer identity with LiveChat"
+desc: "Cross-platform chats for your applications"
+versionGroup: "chat-widget"
+---
+
+# Introduction
+
+## What is it for?
+
+If you have a more advanced service, handling user sessions, you can pair LiveChat customer identity with your own. As a result, your customers will never lose chats they made with your support. What is more, if you also use LiveChat on other platfroms (e.g., in mobile applications), their conversations will sync, making the communication even more seamless and natural. Mind the fact that Customer Identity is not customer metadata like name or email. If you would like to pass those you are probably looking for [updating session variables](https://developers.labs.livechat.com/docs/feature/CW-2556-custom-identity-provider-docs/extending-chat-widget/javascript-api/#update-session-variables).
+
+## Setup
+Customer Identity Provider is essentially a singleton function. You can define as a part of `LiveChatWidget` object from the tracking code used to install ChatWidget on your website or any other place accessing `window.__lc` object, however - **it has to be provided before widget initialization**.
+
+```js
+    // Custom Identity Provider
+    window.__lc.custom_identity_provider = function() {
+      return {
+          getToken: ...,
+          getFreshToken: ...,
+          hasToken: ...,
+          invalidate: ...
+      }
+    }
+```
+
+# Methods
+
+In order to make it work, you'll have to properly implement and provide a set of following methods:
```

Comment:
```suggestion
In order to make Custom Identity Provider work, you'll have to properly implement and provide the following methods:
```

Reply:
It's better to avoid ambiguous "it"

Reply:
I agree on the `it` part, however when it comes to `a set of` I believe it describes better what it is meant to be - a JavaScript object rather than a list or array

Reply:
Okay 👍🏻